### PR TITLE
docs: update common error message in troubleshooting

### PR DESCRIPTION
**Pull request by AnaisUrlichs**
I see several people struggle with this error message

### CODE REVIEW

File: docs/docs/references/troubleshooting.md [hunk 1]
```diff
@@ -12,6 +12,22 @@
...
+    $ trivy image ...
+    ...
+    2024-01-19T08:15:33.288Z	FATAL	image scan error: scan error: unable to initialize a scanner: unable to initialize an image scanner: 4 errors occurred:
+	* docker error: unable to inspect the image (ContainerImageName): Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
+	* containerd error: containerd socket not found: /run/containerd/containerd.sock
+	* podman error: unable to initialize Podman client: no podman socket found: stat podman/podman.sock: no such file or directory
+	* remote error: GET https://index.docker.io/v2/ContainerImageName: MANIFEST_UNKNOWN: manifest unknown; unknown tag=0.1
+    ```
+
+Trivy is unable to access the container image on the remote registry. If you did not specify a registry, Trivy will use the Docker Container Registry.
```
Comment (knqyf263) [inline docs/docs/references/troubleshooting.md:28]: It means Trivy is unable to find the container image in the following places.

- docker daemon
- containerd
- podman
- remote registry

```
trivy image --help | grep image-src
      --image-src strings               image source(s) to use, in priority order (docker,containerd,podman,remote) (default [docker,containerd,podman,remote])
```

The default registry is Docker Hub.

File: docs/docs/references/troubleshooting.md [hunk 2]
```diff
@@ -12,6 +12,22 @@
...
+The container image will likely have to be pushed first to the registry. Then, try to pull the same container image from the registry.
```
Comment (knqyf263) [inline docs/docs/references/troubleshooting.md:29]: Trivy can scan local images.

File: docs/docs/references/troubleshooting.md (hunk 1 above)
Comment (AnaisUrlichs) [reply docs/docs/references/troubleshooting.md:28]: so why did it not scan my container image that I just built prior to running the command?

Comment (knqyf263) [reply docs/docs/references/troubleshooting.md:28]: It depends on your configuration. If your local daemon is not located in the default path, you must specify the path.

```
$ trivy image -h | grep docker-host
      --docker-host string              unix domain socket path to use for docker scanning
```

OR `DOCKER_HOST` also works.
https://docs.docker.com/engine/reference/commandline/cli/

Comment (AnaisUrlichs) [reply docs/docs/references/troubleshooting.md:28]: Oki, thank you, I will update the description

**Pull request closed without merging**